person.rb

```diff
@@ -16,11 +23,38 @@ def can_use_services()
     of_age or @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def of_age()
     return true if @age >= 18
   end
 end
 
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super
+    @nameable = nameable
+  end
 
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] [OPTIONAL] It would be better if you move every class into a separate file that'll makes your projects clean and pro

Reply:
okay. Thank you for the suggestion.